### PR TITLE
refactor: Add new top bar buttons [DRAFT for ACT-1134]

### PR DESCRIPTION
**Pull request by chriskeating603**
### Description

I added a component `TopBarIconButtonV2` which is closer to the solution, but I cannot figure where the height prop is being set. When I change the `borderRadius`, it impacts the width of the ripple, but the height always remains the same. I need to figure out how to set the ripple to be a circle. 

### Related issues

- Fixes [ACT-1134](https://linear.app/valora/issue/ACT-1134/create-new-topbarbuttons-text-icon)

### Network scalability

If a new NetworkId and/or Network are added in the future, the changes in this PR will:

- [ ] Continue to work without code changes, OR trigger a compilation error (guaranteeing we find it when a new network is added)

### CODE REVIEW

**Comment by chriskeating603** [comment]
Ok so I added some changes here where I have different padding for horizontal and vertical. I am using     
`paddingVertical: Spacing.Small12` and `paddingHorizontal: Spacing.Regular16` for the BackButton. I've figured out that this is due to the size of the icon inside, so I think ultimately I want to have the padding amount be responsive to the size of the icon. This way, the shape of the ripple (which is caused by the padding) is always a circle

**Comment by chriskeating603** [comment]
Split out TopBarButton file and TopBarIconButtonV2 + updated BackChevron icon to accommodate new TopBarIconButtonV2 component. Will add any necessary tests on Monday!

File: ios/Podfile.lock [hunk 1]
```diff
@@ -1251,4 +1251,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 81b64061bfc811164c7f86731fc3327e217c8ce4
 
-COCOAPODS: 1.12.1
+COCOAPODS: 1.15.2
```
Comment (MuckT) [inline ios/Podfile.lock:1254]: This looks like an unexpected change.
```suggestion
COCOAPODS: 1.12.1
```

File: src/navigator/TopBarIconButtonV2.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+export type TopBarIconButtonProps = CommonProps & {
+  icon: JSX.Element
+  containerStyle?: ViewStyle
+}
+
+export function TopBarIconButtonV2(props: TopBarIconButtonProps) {
+  return (
+    <Wrapper {...props}>
+      <View style={{ margin: -Spacing.Small12 }}>
```
Comment (MuckT) [inline src/navigator/TopBarIconButtonV2.tsx:62]: Is this needed? A bit strange to have this inline style outside of the containerStyle.

Comment (MuckT) [reply src/navigator/TopBarIconButtonV2.tsx:62]: This should be from components that need it as a container style from `SendSelectRecipient.tsx`

File: src/navigator/TopBarIconButtonV2.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        ? ValoraAnalytics.track(eventName, eventProperties)
+        : ValoraAnalytics.track(eventName)
+    }
+    onPress()
+  }, [onPress, eventName])
+
+  return (
+    <Touchable
+      disabled={disabled}
+      // testID={`${testID}_wrapper`}
```
Comment (MuckT) [inline src/navigator/TopBarIconButtonV2.tsx:43]:
```suggestion
      testID={testID}
```

File: src/navigator/TopBarIconButtonV2.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  containerStyle?: ViewStyle
+}
+
+export function TopBarIconButtonV2(props: TopBarIconButtonProps) {
+  return (
+    <Wrapper {...props}>
+      <View style={{ margin: -Spacing.Small12 }}>
+        <View style={[styles.container, props.containerStyle]}>
+          <Touchable
+            testID={props.testID}
```
Comment (MuckT) [inline src/navigator/TopBarIconButtonV2.tsx:65]:
```suggestion
```

**Comment by MuckT** [comment]
Closing in favor of #5276

**Pull request closed without merging**